src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala

```diff
@@ -1769,18 +1778,29 @@ case class InternalGcsStorage(param: MergedParameters,
         new SFStatement(connection.getSfSession)
       ).getFileTransferMetadatas
         .asScala
-        .map(oneMetadata => result += oneMetadata)
+        .foreach(oneMetadata => {
+          if (!oneMetadata.isForOneFile) {
```

Comment:
`isForOneFile()` is an existing API. It has been enhanced to return true if `down-scoped token` is used.